hardware_interface/include/hardware_interface/components/component_info.hpp

```diff
@@ -24,6 +24,23 @@ namespace hardware_interface
 namespace components
 {
 
+/**
+ * \brief This structure stores information about components defined for a specific hardware
+ * in robot's URDF.
+ */
+struct InterfaceInfo
+{
+  /**
+   * \brief name of the command interfaces that can be set, e.g. "position", "velocity", etc.
+   * Used by joints.
+   */
+  std::string name;
+  /**
+   * \brief (optional) key-value pairs of components parameters, e.g. min_value and max_value.
+   */
+  std::unordered_map<std::string, double> parameters;
```

Comment:
We should leave this type-agnostic on the parser level. I can imagine having interfaces with non-primitive values. So the component should decide which data-type is a parameter.

Also, I propose to limit the interface parameters only to `min` and `max`. Other parameters are anyway probably joint or sensor related. @bmagyar  do you agree?

```suggestion

  /**
   * \brief (optional) minimal allowed values of the interface.
   */
  std::string min_values;
  
  /**
   * \brief (optional) maximal allowed values of the interface.
   */
  std::string max_values;
```